### PR TITLE
service.autostop] 1.0.3

### PR DESCRIPTION
**Pull request by jbinkley60**
### Description
Updated Kodi Matrix addon service.autostop v1.0.3.

This update has the following feature enhancements and fixes:
• Extended sleep timer settings to 180 minutes
• Added pause sleep timer adjust option to select what
to do with the sleep timer when playback is paused.
None - Do nothing and sleep counter continues
Pause - Pause sleep timer
Reset - Reset sleep timer to beginning
• Added 0 -180 secs adjustable GUI sleep timer notification
feature with fixed and adjustable sleep timer extension.
• Added event logging option to aid troubleshooting
• Fixed exception bug when Kodi begins but is unable to play a file properly. Autostop will now ignore those playback errors.

### Checklist:

 -  [X ] My code follows the add-on rules and piracy stance of this project.
 -  [X ] I have read the CONTRIBUTING document
 -  [X ] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: service.autostop/addon.xml [hunk 1]
```diff
@@ -16,6 +16,7 @@
     </assets>
     <summary lang="en_GB">Automatically gives you better control over paused and playing media.</summary>
     <description lang="en_GB">Autostop can automatically stop paused videos and music or set a sleep timer for them to stop playing after a set period of time.  You can also have Autostop automatically start the Kodi screensaver.</description>
-    <source>https://github.com/jbinkley60/service.autostop</source>
+    <forum>http://forum.kodi.tv/showthread.php?tid=211971</forum>
+    <source>https://forum.kodi.tv/showthread.php?tid=365429</source>
```
Comment (enen92) [inline service.autostop/addon.xml:20]: Source is the upstream repo not a forum url

File: service.autostop/common.py [hunk 2]
```diff
@@ -1,10 +1,186 @@
...
+        xbmc.log(mgenlog, xbmc.LOGINFO)
+        dialog = xbmcgui.Dialog()
+        dialog.notification(notifymsg, mgenlog, xbmcgui.NOTIFICATION_INFO, 5000)
+        if settings('screensaver') == 'true':                   #  Active screensaver if option selected
+            xbmc.executebuiltin('ActivateScreensaver')
+        settings('notifyset', 'no')                             # Clear notification flag
+        settings('varextnotify', 'no')                          # Clear notification flag
+    except:
+        xbmc.log('Autostop error when stopping playback.', xbmc.LOGINFO)        
+        pass
```
Comment (enen92) [inline service.autostop/common.py:162]: No need for pass

File: service.autostop/common.py [hunk 3]
```diff
@@ -1,10 +1,186 @@
...
+        if asevlog == 'true':
+            xbmc.log('Autostop calculating extension time. ' + str(currpos) + ' ' +     \
+            str(endpos) + ' ' + str(plcount) + ' ' + str(remaintime) + ' ' + extension  \
+            + ' ' + str(plstoptime * 60), xbmc.LOGINFO)
+        settings('varextnotify', 'yes')                         # Only display progress bar once  
+        return extension
+    except:
+        xbmc.log('Autostop error getting remaining extension time.', xbmc.LOGINFO)
+        return '-1'
+        pass
```
Comment (enen92) [inline service.autostop/common.py:184]: Same

File: service.autostop/service.py [hunk 4]
```diff
@@ -54,66 +59,48 @@ def onPlayBackStopped(self):
...
-                    ptitle = ptag.getTitle()
-                elif xbmc.Player().isPlayingAudio():
-                    ptag = xbmc.Player().getMusicInfoTag()
-                    ptitle = ptag.getTitle()                
-                else:
-                    ptitle = "playing file"
                 pos = xbmc.Player().getTime()
-                xbmc.Player().stop()
+                logmsg = 'Autostop stopped paused playback: '
+                notifymsg = 'Autostop Paused Timer'
```
Comment (enen92) [inline service.autostop/service.py:70]: If on the GUI the strings needs to be localized

File: service.autostop/service.py [hunk 5]
```diff
@@ -54,66 +59,48 @@ def onPlayBackStopped(self):
...
+            extime = float(settings('extime'))
+            asevlog = settings('asevlog')
+            totalstoptime = plstoptime + extime
+            if asevlog == 'true':
+                xbmc.log('Autostop play count and stop time ' + str(plcount) + ' ' +         \
+                str(plstoptime) + ' ' + str(player.plflag), xbmc.LOGINFO)
+            if plstoptime > 0 and plcount >= totalstoptime * 60 and (player.plflag > 0 or    \
+            (player.plflag == 0 and player.paflag == 1)):
+                logmsg = 'Autostop stopped current playback: '
+                notifymsg = 'Autostop Sleep Timer'
```
Comment (enen92) [inline service.autostop/service.py:97]: Same

File: service.autostop/service.py [hunk 6]
```diff
@@ -54,66 +59,48 @@ def onPlayBackStopped(self):
...
                 plcount = 0
+                stopPlayback(notifymsg, logmsg)      # Stop playback if playing too long
         except:
-            xbmc.log('Autostop play count and stop time exception error' + str(pacount) + ' ' + str(pastoptime) +    \
-            ' ' + str(player.paflag), xbmc.LOGINFO)         
+            xbmc.log('Autostop play count and stop time exception error' + str(pacount) + ' ' \
+            + str(pastoptime) + ' ' + str(player.paflag), xbmc.LOGINFO)
+            pass
```
Comment (enen92) [inline service.autostop/service.py:103]: Remove pass if possible

File: service.autostop/addon.xml (hunk 1 above)
Comment (jbinkley60) [reply service.autostop/addon.xml]: Thanks for catching.  No idea how I changed this.  It is fixed.

File: service.autostop/common.py (hunk 2 above)
Comment (jbinkley60) [reply service.autostop/common.py]: Removed.

File: service.autostop/common.py (hunk 3 above)
Comment (jbinkley60) [reply service.autostop/common.py]: Removed

File: service.autostop/service.py (hunk 4 above)
Comment (jbinkley60) [reply service.autostop/service.py]: This has been updated to localized strings.

File: service.autostop/service.py (hunk 5 above)
Comment (jbinkley60) [reply service.autostop/service.py]: This has been updated to localized strings.

File: service.autostop/service.py (hunk 6 above)
Comment (jbinkley60) [reply service.autostop/service.py]: Removed.

**Review by enen92 — APPROVED** [review]

**Pull request merged**